oak_functions/sdk/oak_functions/tests/module/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Oak Functions ABI test.
+
+use anyhow::{anyhow, Context};
+use assert_matches::assert_matches;
+use maplit::hashmap;
+use std::collections::HashMap;
+
+type TestFn = fn(&str) -> ();
+
+struct TestManager<'a> {
+    tests: HashMap<&'a str, TestFn>,
+}
+
+impl TestManager<'static> {
+    fn new() -> Self {
+        Self {
+            tests: hashmap! [
```

Comment:
I was thinking about this, I think once we have a way of specifying something like RPCs over channels, we could abstract this away via something like tarpc, with a method for each test, and generate client and server stubs automatically. Would that work?

Reply:
I don't see why not! I'll keep it in mind!